packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -705,6 +708,16 @@ class ScrollableState<T extends Scrollable> extends State<T> with SingleTickerPr
     });
   }
 
+  // Used for state changes that sometimes occur during a build phase. If so,
+  // we skip calling setState, as the changes will apply to the next build.
+  void _setStateMaybeDuringBuild(VoidCallback fn) {
+    if (SchedulerBinding.instance.schedulerPhase == SchedulerPhase.persistentCallbacks) {
```

Comment:
Yeah, it's ugly and I'm hoping it won't be necessary in the Scrollable rewrite. I'll add a note in the comments.